.github/workflows/blocked-prs.yml

```diff
@@ -0,0 +1,245 @@
+                "prNumber": .number,
+                "prHeadSha": .head.sha,
+                "prHeadLabel": .head.label,
+                "prBody": .body,
+                "prLabels": (reduce .labels[].name as $l ([]; . + [$l]))
+              }
+            ' <<< "$PR_JSON")"
+          } >> "$GITHUB_ENV"
+
+      - name: Find Blocked/Stacked PRs in body
+        id: pr_ids
+        run: |
+          prs=$(
+            jq -c '
+              .prBody as $body 
+                | (
+                    $body | 
+                      reduce ( 
+                        . | scan("blocked (?:by|on):? #([0-9]+)") 
+                        | map({
+                          "type": "Blocked on",
+                          "number": ( . | tonumber )
+                        })
+                      ) as $i ([]; . + [$i[]])
+                  ) as $bprs
+                | (
+                    $body | 
+                      reduce ( 
+                        . | scan("stacked on:? #([0-9]+)")
+                        | map({
+                          "type": "Stacked on",
+                          "number": ( . | tonumber )
+                        })
+                      ) as $i ([]; . + [$i[]])
+                  ) as $sprs
+                | ($bprs + $sprs) as $prs 
+                | { 
+                    "blocking": $prs,
+                    "numBlocking": ( $prs | length),
+                  }
+              ' <<< "$JOB_DATA"
+          )
+          echo "prs=$prs" >> "$GITHUB_OUTPUT"
+
+      - name: Collect Blocked PR Data
+        id: blocking_data
+        if: fromJSON(steps.pr_ids.outputs.prs).numBlocking > 0
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BLOCKING_PRS: ${{ steps.pr_ids.outputs.prs }}
+        run: |
+          blocked_pr_data=$(
+            while read -r pr_data ; do
+              gh api \
+                -H "Accept: application/vnd.github+json" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                "/repos/$OWNER/$REPO/pulls/$(jq -r '.number' <<< "$pr_data")" \
+                | jq -c --arg type "$(jq -r '.type' <<< "$pr_data")" \
+                  '
+                    . | {
+                      "type": $type,
+                      "number": .number,
+                      "merged": .merged,
+                      "labels": (reduce .labels[].name as $l ([]; . + [$l])), 
+                      "basePrUrl": .html_url,
+                      "baseRepoName": .head.repo.name,
+                      "baseRepoOwner": .head.repo.owner.login,
+                      "baseRepoUrl": .head.repo.html_url,
+                      "baseSha": .head.sha,
+                      "baseRefName": .head.ref,
+                    }
+                  '
+            done < <(jq -c '.blocking[]' <<< "$BLOCKING_PRS") | jq -c -s
+          )
+          blocked_by_labels=$(jq -c 'map( select( .merged | not ) | "blocked-by:" + (.number | tostring))' <<< "$blocked_pr_data" )
```

Comment:
I don't think we really need this. The comment and blocked label should do enough, and on larger stacks this will get kinda spammy. It also create a fair bit more complication and I would *very* much like to see this slimmed down where possible

Reply:
if there is a large stack each pr in the stack should only really have at most two labels. that said. I could force the PR back to the state where it only used two labels if that's preferable.